src/tools/utils.rs

```diff
@@ -115,3 +119,53 @@ where
     );
     Ok(())
 }
+
+/// block on function
+pub fn block_on<T, F: Future<Output = T>>(f: F) -> Result<T> {
```

Comment:
I think, this function should not be a public

Reply:
It is also used in tools/validation/validate_index()

Reply:
Ok. Didn't notice that